docs/learn-glossary.md

```diff
@@ -3,160 +3,175 @@ id: learn-glossary
 
-A client-side proxy used to make remote invocations to an entity that it
-represents. For example, to start a workflow, a stub object that represents
-this workflow is created through a special API. Then this stub is used to start,
-query, or signal the corresponding workflow.
+Archival is a feature that automatically moves [Event Histories](#event-history) from normal persistence to a blob store after the [Workflow](#workflow) retention period.
 
-The Go client doesn't use this.
+- The purpose of Archival is to be able to keep [Event Histories](#event-history) as long as needed while not overwhelming the persistence store.
+- There are two reasons why you may want to keep [Event Histories]((#event-history)) after the retention period has passed:
+  1. Compliance: For legal reasons, [Event Histories](#event-history) may need to be stored for a long period of time.
+  2. Debugging: Older [Event Histories](#event-history) can be referenced to help with debugging.
 
-### Decision
+### Client Stub
 
-Any action taken by the workflow durable function is called a decision. For example:
-scheduling an activity, canceling a child workflow, or starting a timer. A [decision task](#decision-task) contains an optional list of decisions. Every decision is recorded in the [event history](#event-history) as an [event](#event).
+A Client Stub is a client-side proxy in the Java SDK which is used to make remote invocations on an entity that it represents.
 
-### Decision Task
+- To start a [Workflow](#workflow), for example, a Stub object which represents the [Workflow](#workflow) is created through a special API. Then the Stub is used to start, query, or signal the corresponding [Workflow](#worker).
+- The Go SDK does make use of a Client Stub.
 
-Every time a new external event that might affect a workflow state is recorded, a decision task that contains it is added to a [decision-task-queue](#decision-task-queue) and then picked up by a workflow worker. After the new event is handled, the decision task is completed with a list of [decisions](#decision).
-Note that handling of a decision task is usually very fast and is not related to duration
-of operations that the workflow invokes.
+### Command
 
-### Decision Task Queue
+Any action taken by the [Workflow](#workflow) durable function is called a Command.
 
-Task queue that is used to deliver [decision tasks](#decision-task) to [workflow workers](#workflow-worker)
+- Scheduling an [Activity](#activity), canceling a child [Workflow](#workflow), or starting a timer are all Commands for example.
+- A [Workflow Task](#workflow-task) contains an optional list of Commands.
+- Every Command is recorded in the [Event History](#event-history) as an [Event](#event).
 
 ### Event
 
-An indivisible operation performed by your application. For example,
-activity_task_started, task_failed, or timer_canceled. Events are recorded in the event history.
+An Event is an indivisible operation performed by your application.
```

Comment:
I am not sure but may be you should highlight term somehow like:
```suggestion
An `Event` is an indivisible operation performed by your application.
```

Reply:
I would like to reserve the backtacks (which style text to look like code) for more technical labels/references - But, I think we can achieve this by making concepts bold (at least in the glossary).